client/AI/AIClientApp.cpp

```diff
@@ -401,7 +401,11 @@ void AIClientApp::HandleMessage(const Message& msg) {
 
     case Message::CHECKSUM: {
         TraceLogger() << "(AIClientApp) CheckSum.";
-        VerifyCheckSum(msg);
+        bool result = VerifyCheckSum(msg);
+        if (!result) {
+            ErrorLogger() << "Wrong checksum";
+            throw std::runtime_error("AI got incorrect checksum.");
```

Comment:
Is this caught somewhere or do you intend to just crash / kill the AI if it has a bad checksum...?

Reply:
I intend to kill AI.

Reply:
Have you had cases where an AI had a bad checksum? They usually (only?) run on the same machine as the server, I think... so wouldn't expect this to happen.

Would this kill them when setting up a multiplayer game, so attempting to add a buggy AI will fail and drop the connection?

Presumably this would break starting single-player games in the event that an AI got a bad checksum somehow.

Reply:
@geoffthemedio No yet but in case of using alternative content we are testing now it possible. I also remember having issues with checksum in tests which are definitely run on the same machine as the server.

AI launches at game start which will be failed.